scripts/dev/generate_embeddable_epJSON_schema/generate_embeddable_epJSON_schema.cpp

```diff
@@ -16,7 +16,7 @@ int main( int argc, char const *argv[] )
 		std::cout << "schema file path " << argv[ 1 ] << " not found" << std::endl;
 		return 1;
 	}
-	auto const input_json = json::parse( schema_stream );
+	auto const &input_json = json::parse( schema_stream );
```

Comment:
A branch for discussion of correct use of auto

Reply:
Uber-point: the correct use of auto is to use it as little as possible. In my opinion, it serves no positive function other than to make programmers feel like they are cool. And it makes the code less readable to other people.

I am okay with limited use of `auto` for iterators and `auto (const) &` for shortcut object references. Every other use is just annoying. 

That said, I actually don't know what to say about this one. For one thing, I don't know what kind of object is returned by `json::parse`. For another, I don't know whether the caller owns the return value (in which case it has to be copy constructed or move-constructed, i.e., not a reference) or whether the json library creates a persistent object such that holding only a reference to it is okay. I would leave this instance alone.